src/main/java/org/mondemand/Context.java

```diff
@@ -26,33 +28,25 @@ public String getKey() {
     return key;
   }
 
-  /**
-   * @param key the key to set
-   */
-  public void setKey(String key) {
-    this.key = key;
-  }
-
   /**
    * @return the value
    */
   public String getValue() {
     return value;
   }
 
-  /**
-   * @param value the value to set
-   */
-  public void setValue(String value) {
-    this.value = value;
-  }
-
-  public Context() {
-  }
-
-  public Context(String k, String v) {
+  public Context(String k, String v) throws MondemandException {
     key = k;
     value = v;
+    if(key == null) {
```

Comment:
this constructor is called when applications use ContextList. Are we sure a context with null key/value should be rejected?

Reply:
was it not the null value that was causing the mondemand server to crash?

Reply:
It was actually the missing values, in that I believe somewhere during serialization it's dropping values that are null and then the indices are off.  So this can be solved in a number of ways, but the requirement is that the number of keys and values for contexts which are serialized in the event are equal.  So the crash was 3 keys and 2 values serialized, that shouldn't happen.  We'll also fix on the receive side but that won't help as it's disconnected from the send, so we want some sort of fix here.